### PR TITLE
Add missing attributes to google_container_cluster resource

### PR DESCRIPTION
**Pull request by soerenmartius**

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -179,8 +292,12 @@ resource "google_container_cluster" "cluster" {
...
   }
 
-  workload_identity_config {
-    workload_pool = "${var.project}.svc.id.goog"
+  dynamic "workload_identity_config" {
+    for_each = !var.enable_autopilot ? [1] : []
+
+    content {
+      workload_pool = "${local.project}.svc.id.goog"
+    }
```
Comment (mariux) [inline main.tf:300]: why does this depend on autopilot?

File: main.tf [hunk 2]
```diff
@@ -154,11 +261,14 @@ resource "google_container_cluster" "cluster" {
...
 
-  ### private cluster config
+  # --------------------------------------------------------------------------------------------------------------------
+  # PRIVATE CLUSTER CONFIG
   # From the provider documentation:
-  #   The Google provider is unable to validate certain configurations of private_cluster_config
-  #   when enable_private_nodes is false.
-  #   It's recommended that you omit the block entirely if the field is not set to true.
+  #   - The Google provider is unable to validate certain configurations of private_cluster_config
+  #   - When enable_private_nodes is false.
```
Comment (mariux) [inline main.tf:268]: that is not 2 points?

File: main.tf [hunk 3]
```diff
@@ -108,39 +170,84 @@ resource "google_container_cluster" "cluster" {
...
   # there is no way not to create the default node pool.
   # node pools should be created using the terraform-google-gke-node-pool module
-  # for regional clusters thsi will create 1 node in each zone of the region
+  # for regional clusters that will create 1 node in each zone of the region
+  # --------------------------------------------------------------------------------------------------------------------
 
-  # node_pool {}
   initial_node_count       = 1
-  remove_default_node_pool = true
+  remove_default_node_pool = !var.enable_autopilot ? true : null
```
Comment (mariux) [inline main.tf:245]: where is this relation to autopilot described in detail?

**Review by mariux — CHANGES_REQUESTED** [review]
can we add all autopilot related changes into a single space by creating locals based on the mode...  we once agreed that we do not ignore user set values without notice and rather make it fail to give the feedback. by silently ignoring we create unexpected behavior. i would rather vote for making it fail or for creating a autopilot version as most of the config seems to be invalid anyway when autopilot is used.

File: main.tf [hunk 4]
```diff
@@ -1,7 +1,14 @@
+data "google_project" "project" {}
+
+locals {
+  project = var.project != null ? var.project : data.google_project.project.id
```
Comment (mariux) [inline main.tf:4]: that is the terraform id... please use project_id

File: main.tf (hunk 1 above)
Comment (soerenmartius) [reply main.tf:300]: because autopilot doesn't support `workload_identity_config`

File: main.tf (hunk 4 above)
Comment (soerenmartius) [reply main.tf:4]: what is a terraform id?

File: main.tf (hunk 1 above)
Comment (soerenmartius) [reply main.tf]: removed since we will move autopilot to a different module at some point

File: main.tf (hunk 2 above)
Comment (soerenmartius) [reply main.tf]: resolved

File: main.tf (hunk 3 above)
Comment (soerenmartius) [reply main.tf]: resolved

Comment (soerenmartius) [reply main.tf]: removed since we will move autopilot to a different module at some point

**Comment by soerenmartius** [comment]
@mariux resolved the points apart of the project since I am not sure what you mean. I am gonna create a separate autopilot module once we need it.

File: main.tf (hunk 4 above)
Comment (mariux) [reply main.tf:4]: `.id` is terraform internal ID.... `.project_id` is what you need to use..

**Review by mariux — CHANGES_REQUESTED** [review]

**Comment by soerenmartius** [comment]
@mariux ready 4 review again :)

File: CHANGELOG.md [hunk 5]
```diff
@@ -22,6 +22,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - BREAKING CHANGE: Removed `master_auth` block
 - Refactored `workload_pool` block
 
+### Changed
+
+- Remove support for `google_project_service` resource
```
Comment (mariux) [inline CHANGELOG.md:27]: this is breaking.

**Review by mariux — CHANGES_REQUESTED** [review]
thanks for the PR.. 

As the scope is quiet big and multiple changes are ntroduced here, please split this PR into refactor, template updates and new features so we can process reviews easier.. There are conflicts in the scope and duplicates in the CHANGELOG.

**Pull request closed without merging**